### ISSUE TITLE
"go get github.com/0xERR0R/blocky" fetches v0.9.1

### ISSUE THREAD

**Issue by markdrayton** [0xERR0R/blocky#1594]
tl;dr: `go get -x github.com/0xERR0R/blocky` fetches v0.9.1, not the most recent version (v0.24 today).

I'm trying to add blocky to a [gokrazy](https://gokrazy.org/) installation. gokrazy uses modules to do this (see https://gokrazy.org/packages/). I added blocky to my instance:

```
$ gok add github.com/0xERR0R/blocky
2024/09/04 22:29:24 Adding github.com/0xERR0R/blocky as a (non-local) package to gokrazy instance gokrazy
2024/09/04 22:29:24 Adding the following package to gokrazy instance "gokrazy":
  Go package  : github.com/0xERR0R/blocky
  in Go module: github.com/0xERR0R/blocky
2024/09/04 22:29:24 Creating gokrazy builddir for module github.com/0xERR0R/blocky
2024/09/04 22:29:24 Creating go.mod based on upstream go.mod
2024/09/04 22:29:24 Adding package to gokrazy config
```

but when I try to update it I get:

```
$ gok -i gokrazy update
[..]
[building (go compiler)]go: github.com/0xERR0R/blocky@v0.9.1: parsing go.mod:
	module declares its path as: blocky
	        but was required as: github.com/0xERR0R/blocky
2024/09/04 22:31:22 getting incomplete packages [github.com/0xERR0R/blocky]
```

The issue here seems to be that `gok` fetched v0.9.1, which has a [very old go.mod](https://github.com/0xERR0R/blocky/blob/d19dd7d45e7d23a74f64cccce0e856e64cc99136/go.mod), instead of the latest version. This also reproduces with `go get`:

```
$ cd ~/go/src
$ mkdir test; cd test
$ go mod init
go: creating new go.mod: module test
$ go get -x github.com/0xERR0R/blocky
# get https://proxy.golang.org/github.com/@v/list
# get https://proxy.golang.org/github.com/0x%21e%21r%21r0%21r/@v/list
# get https://proxy.golang.org/github.com/0x%21e%21r%21r0%21r/blocky/@v/list
# get https://proxy.golang.org/github.com/0x%21e%21r%21r0%21r/blocky/@v/list: 200 OK (0.080s)
# get https://proxy.golang.org/github.com/@v/list: 404 Not Found (0.144s)
# get https://proxy.golang.org/github.com/0x%21e%21r%21r0%21r/@v/list: 404 Not Found (0.144s)
go: github.com/0xERR0R/blocky@upgrade (v0.9.1) requires github.com/0xERR0R/blocky@v0.9.1: parsing go.mod:
	module declares its path as: blocky
	        but was required as: github.com/0xERR0R/blocky
```

I don't quite know what's going on here — I'm far from an expert in Go modules and my casual digging around in the source code didn't turn anything up. It's suspicious that v0.9.1 lexicographically sorts last, and also that the proxy only seems to have versions with `major.minor.patch` numbering:

```
$ curl 'https://proxy.golang.org/github.com/0x%21e%21r%21r0%21r/blocky/@v/list'
v0.3.5
v0.2.1
v0.9.1
v0.2.0
```

is blocky's module versioning at fault? I couldn't find any clear suggestion that it is. Is this a known Go modules problem? I couldn't find anything here either :-(

**Comment by ThinkChaos** [comment]
Hi, I believe this is because the tags are not semantic versions, which Go modules expect.
 
This would be solved by #1505.  
In the mean time, `github.com/0xERR0R/blocky@3ab04562fe1e887799bfcc85a116e12e79643694` should work (that's the commit v0.24 points to) but you'll need to do manual updates.  
Alternatively you could do `@main` to get updates, but use a non release.

Unfortunately, `@v0.24` doesn't work because Go tries to interpret that as a version and no matching version exists because of #1505.

I'll close this in favor of #1505, we can continue discussing there, gokrazy was even already mentioned! :)

**Comment by markdrayton** [comment]
Thanks. I initially suspected semver but talked myself out of it when I found `semver.Compare("v0.9.1", "v0.24")` worked as expected. I'll follow #1505!
